point_cloud_transport/src/publisher_plugin.cpp

```diff
@@ -38,12 +38,6 @@
 namespace point_cloud_transport
 {
 
-PublisherPlugin::EncodeResult PublisherPlugin::encode(const sensor_msgs::msg::PointCloud2 & raw)
-const
-{
-  return this->encode(raw);
-}
-
```

Comment:
Oh, I see.  Yeah, clang is totally right about the old code; any call to this function will just have it call itself.  What we really wanted is to have this as a pure virtual function, which this change now does.  So I think this is good now.